source/Octopus.Cli/Commands/CreateChannelCommand.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using log4net;
+using Octopus.Cli.Infrastructure;
+using Octopus.Cli.Util;
+using Octopus.Client.Model;
+
+namespace Octopus.Cli.Commands
+{
+    [Command("create-channel", Description = "Creates a channel for a project")]
+    public class CreateChannelCommand : ApiCommand
+    {
+        public CreateChannelCommand(IOctopusRepositoryFactory repositoryFactory, ILog log, IOctopusFileSystem fileSystem) : base(repositoryFactory, log, fileSystem)
+        {
+            var options = Options.For("Create");
+            options.Add("channel=", "The name of the channel to create", s => channelName = s);
+            options.Add("description=", "A description of the channel", d => channelDescription = d);
+            options.Add("project=", "The name of the project in which to create the channel", p => projectName = p);
+            options.Add("lifecycle=", "The name of the lifecycle with which to create the channel", l => lifecycleName = l);
+            options.Add("make-default-channel", "Set the new channel to be the default channel", _ => makeDefaultChannel = true);
+            options.Add("update-if-exists", "Updates the channel if it already exists", _ => updateIfExists = true);
+        }
+
+        string channelName;
+        string projectName;
+        string lifecycleName;
+        string channelDescription;
+        bool updateIfExists;
+        bool? makeDefaultChannel;
+
+        protected override void Execute()
+        {
+            if (string.IsNullOrWhiteSpace(projectName)) throw new CommandException("Please specify a project using the parameter: --project=ProjectXYZ");
+            if (string.IsNullOrWhiteSpace(channelName)) throw new CommandException("Please specify a channel name using the parameter: --channel=ChannelXYZ");
+            if (string.IsNullOrWhiteSpace(lifecycleName)) throw new CommandException("Please specify a lifecycle name using the parameter: --lifecycle=LifecycleXYZ");
```

Comment:
A `null` lifecycle is allowed, which is interpreted as "Use the project's default lifecycle". I think for consistency `octo.exe` should allow the same behaviour as the UI.